lib/hammer_cli_foreman_admin/logging_command.rb

```diff
@@ -0,0 +1,127 @@
+
+    option ["-d", "--level-debug"], :flag, "Increase verbosity level to debug."
+    option ["-p", "--level-production"], :flag, "Decrease verbosity level to standard."
+    option ["-c", "--components"], "COMPONENTS", "Components to apply, use --list to get them.", :format => HammerCLI::Options::Normalizers::List.new
+    option ["-l", "--list"], :flag, "List available components."
+    option ["-a", "--all"], :flag, "Apply to all components."
+    option ["-n", "--dry-run"], :flag, "Do not apply specified changes."
+    option "--no-backup", :flag, "Skip configuration backups creation."
+
+    validate_options do
+      any(:option_level_debug, :option_level_production, :option_list).required
+      any(:option_components, :option_all, :option_list).required
+      all(:option_level_debug, :option_level_production).rejected
+      any(:option_level_debug, :option_level_production).rejected if option(:option_list).exist?
+    end
+
+    def run_command(cmd, raise_on_error = true, verbose = false)
+      output = `#{cmd}`
+      print_message output if verbose && output.chomp != ''
+      raise("return value = #{$?.to_i}") if $?.to_i != 0 && raise_on_error
+    rescue Exception => e
+      if raise_on_error
+        raise "Command '#{cmd}' failed: #{e}"
+      else
+        print_message "Command '#{cmd}' failed: #{e}"
+      end
+    end
+
+    def check_options(hash, action_name, *required_keys)
+      required_keys.each do |key|
+        raise("Missing option '#{key}' in action '#{action_name}' component '#{hash[:name]}'") unless hash[key]
+      end
+    end
+
+    def action_functions
+      @action_functions ||= {
+        :run_command => lambda do |opts|
+          check_options opts, :run_command, :command
+          run_command opts[:command], false
+        end,
+        :run_command_on_file => lambda do |opts|
+          check_options opts, :run_command_on_file, :command, :file
+          run_command "#{opts[:command]} #{opts[:file]}", false
+        end,
+        :create_file => lambda do |opts|
+          check_options opts, :create_file, :file, :contents
+          FileUtils.mkdir_p(File.dirname(opts[:file]))
+          open(opts[:file], 'w') { |f| f.puts opts[:contents] }
+        end,
+        :remove_file => lambda do |opts|
+          check_options opts, :remove_file, :file
+          File.unlink(opts[:file])
+        end,
+        :ensure_line_is_present => lambda do |opts|
+          check_options opts, :ensure_line_is_present, :file, :line
+          if File.foreach(opts[:file]).grep(/#{opts[:line][0]}/).empty?
+            open(opts[:file], 'a') { |f| f.puts "\n" + opts[:line].join }
+          else
+            run_command %Q|sed -i 's!#*#{opts[:line][0]}\s*#{opts[:line][1]}.*!#{opts[:line].join}!' #{opts[:file]}|
+          end
+        end
+      }
+    end
+
+    def new_level
+      option_level_debug? ? :debug : :production
+    end
+
+    def configure_component(component, level)
+      name = component[:name]
```

Comment:
Could you add something like `check_options` here too, please? When I tried to edit the settings and misconfigured them, I usually end up with `NoMethodError (undefined method `[]' for nil:NilClass)`